src/session_impl.cpp

```diff
@@ -1897,7 +1904,7 @@ namespace aux {
 					if (m_alerts.should_post<listen_failed_alert>())
 					{
 						m_alerts.emplace_alert<listen_failed_alert>(device
-							, tcp::endpoint(), listen_failed_alert::enum_if, ec
+							, address(), 0, listen_failed_alert::enum_if, ec
```

Comment:
and maybe there even should be a constructor that doesn't take an (address,port) pair